crates/sargon/src/profile/mfa/security_structures/matrices/builder/violation_to_error_conversion.rs

```diff
@@ -0,0 +1,62 @@
+use crate::prelude::*;
+
+impl From<MatrixRolesInCombinationBasicViolation> for CommonError {
```

Comment:
The idea of `CommonError` is to be flat. Always. And by one single error code know what went wrong. But yeah it is hard to do that. Error handling in general is temporary. And we are no somewhat "stuck" with what was meant to be temporary. And now we have bigger fish to fry. We also note that, unfortunately, we are somewhat limit to UniFFI when it comes to cool stuff like `backtrace` support for errors (search for "backtrace": https://docs.rs/thiserror/latest/thiserror/ )

Anyhow [pun not intended](https://docs.rs/anyhow/latest/anyhow/))... this part of the PR is a bit in flux, I'm working on the SecurityShieldBuilder in `sargon` and might not include so many errors at all in CommonError, since most errors we can actually **prevent** by letting `SecurityShieldBuilder` preventing them and thus UI.